pkg/client/unversioned/helper_blackbox_test.go

```diff
@@ -42,50 +43,50 @@ func objBody(object interface{}) io.ReadCloser {
 
 func TestNegotiateVersion(t *testing.T) {
 	tests := []struct {
-		name, version, expectedVersion string
-		serverVersions                 []string
-		clientVersions                 []string
-		config                         *unversioned.Config
-		expectErr                      func(err error) bool
-		sendErr                        error
+		name            string
+		version         *uapi.GroupVersion
+		expectedVersion *uapi.GroupVersion
+		serverVersions  []string
+		clientVersions  []uapi.GroupVersion
+		config          *unversioned.Config
+		expectErr       func(err error) bool
+		sendErr         error
 	}{
 		{
 			name:            "server supports client default",
-			version:         "version1",
+			version:         &uapi.GroupVersion{Version: "version1"},
 			config:          &unversioned.Config{},
-			serverVersions:  []string{"version1", testapi.Default.Version()},
-			clientVersions:  []string{"version1", testapi.Default.Version()},
-			expectedVersion: "version1",
+			serverVersions:  []string{"/version1", testapi.Default.Version()},
```

Comment:
ideally, this should be a valid version, either "v1" or "testgroup/version1". add a TODO to the cleanup list to prevent parsing empty groups like this and cleanup tests like this